src/Store.cpp

```diff
@@ -14,92 +14,210 @@ const unsigned short bombCost = 100;
+		cout << " 0) Exit" << endl << endl;
+		cout << " ---------------------------------- " << endl << endl;
+		cout << " What would you like to do today? " << endl << endl << " ";
+		choice = input();
+		cout << endl;
+
+		isValidChoice = true;
+		switch (choice)
+		{
+		case 1: // Buy
+			buyItems = true;
+			break;
+		case 2: // Sell
+			buyItems = false;
+			break;
+		case 0: // Exit
+			break;
+		default:
+			cout << invalidChoice << endl;
+			isValidChoice = false;
+			Sleep(SLEEP_MS);
+			break;
+		}
+	} while (!isValidChoice);
+
+	// Allow player to buy/sell items as long as the player does not want to exit the store
+	while (choice != 0)
+	{
+		ClearScreen();
+		cout << endl << endl;
+		_Player->DisplayInventory();
+		cout << endl << " Jeremy's Tools Store" << endl << endl;
 		cout << " ----ITEM----------------COST------ " << endl << endl;
 		cout << " 1) Arrows x 5\t\t " << arrowsCost << endl;
 		cout << " 2) Bomb\t\t " << bombCost << endl;
 		cout << " 3) Potion\t\t " << potionCost << endl;
 		cout << " 4) Whetstone\t\t " << whetstoneCost << endl << endl;
 		cout << " 0) Exit" << endl << endl;
 		cout << " ---------------------------------- " << endl << endl;
-		cout << " What do you want to buy today?" << endl << endl << " ";
+		if (buyItems)
+		{
+			cout << " What do you want to buy today?" << endl << endl << " ";
+		}
+		else
+		{
+			cout << " What do you want to sell today?" << endl << endl << " ";
+		}
 		choice = input();
 		cout << endl;
+
 		switch (choice)
 		{
 		case ITEMTYPE::ARROWS:
-			if (coins < arrowsCost)
-			{
-				cout << lowCoinsMessage << endl;
-			}
-			else
-			{
-				cout << " 5 Arrows purchased ! " << endl;
-				_Player->AddStoreItemToInventory(ITEMTYPE::ARROWS);
-				_Player->LoseCoins(arrowsCost);
-			}
-			Sleep(SLEEP_MS);
-			break;
 		case ITEMTYPE::BOMB:
-			if (coins < bombCost)
-			{
-				cout << lowCoinsMessage << endl;
-			}
-			else
-			{
-				cout << " Bomb purchased !" << endl;
-				_Player->AddStoreItemToInventory(ITEMTYPE::BOMB);
-				_Player->LoseCoins(bombCost);
-			}
-			Sleep(SLEEP_MS);
-			break;
 		case ITEMTYPE::POTION:
-			if (coins < potionCost)
-			{
-				cout << lowCoinsMessage << endl;
-			}
-			else
-			{
-				cout << " Potion purchased !" << endl;
-				_Player->AddStoreItemToInventory(ITEMTYPE::POTION);
-				_Player->LoseCoins(potionCost);
-			}
-			Sleep(SLEEP_MS);
-			break;
 		case ITEMTYPE::WHETSTONE:
-			if (coins < whetstoneCost)
+			if (buyItems)
 			{
-				cout << lowCoinsMessage << endl;
+				BuyItem(_Player, choice);
 			}
 			else
 			{
-				cout << " Whetstone purchased !" << endl;
-				_Player->AddStoreItemToInventory(ITEMTYPE::WHETSTONE);
-				_Player->LoseCoins(whetstoneCost);
+				SellItem(_Player, choice);
 			}
 			Sleep(SLEEP_MS);
-			break;
-		case 0://Exit Store
+		case 0: // Exit
 			break;
 		default:
 			cout << invalidChoice << endl;
 			Sleep(SLEEP_MS);
 		}
-	} while (choice);
+	}
+}
+
+void Store::SellItem(Player* _Player, int _Item) {
```

Comment:
There is a lot of code duplication here and in `BuyItem()`, the `cout` messages in particular are repeated a lot ("Arrows sold", "Bomb sold", etc. and the `notEnoughItemsMessage`) Is there any way to do this more efficiently? Maybe with a hash map?

Reply:
Also, maybe `SellItem`, `BuyItem` and `RemoveStoreItemFromInventory` should include another parameter `int amount`, which specifies the number of items to buy/sell, instead of hardcoding the amount.

Reply:
I grouped certain properties of a basic item in a struct (1d9bb14c7be04b9eb871183b97c1962a3f17e5da). The info structure contains the amount as well and uses it to compute the total price.